main_task/src/main/Student.java

```diff
@@ -0,0 +1,74 @@
+package main;
+
+public class Student implements Comparable<Student> {
+    private String firstName;
+    private String lastName;
+    private int age;
+    private Student previousStudent;
+    private Student nextStudent;
+
+    public Student getPreviousStudent() {
+        return previousStudent;
+    }
+
+    public void setPreviousStudent(Student previousStudent) {
+        this.previousStudent = previousStudent;
+    }
+
+    public Student getNextStudent() {
+        return nextStudent;
+    }
+
+    public void setNextStudent(Student nextStudent) {
+        this.nextStudent = nextStudent;
+    }
+
+    public Student(String firstName, String lastName, int age) {
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.age = age;
+    }
+
+    public String getFirstName() {
+        return firstName;
+    }
+
+    public String getLastName() {
+        return lastName;
+    }
+
+    public int getAge() {
+        return age;
+    }
+
+    @Override
+    public String toString() {
+        return lastName + " " + firstName + " " + age;
+    }
+
+    @Override
+    public int compareTo(Student s) {
+        return this.lastName.compareTo(s.getLastName());
+    }
+
+    @Override
+    public boolean equals(Object s) {
+        Student tempStudent = (Student) s;
+
+        if (s == this) {
+            return true;
+        }
+        if (this.firstName != tempStudent.firstName) {
```

Comment:
using != will work in case of this case, because the firstName is initialized with a string literal, e.g.,  String firstName ="Hakob". If it is already present in the memory, then Java returns the same instance of string object instead of creating a new String object with the same value. Things are different if a String object is created with new keyword. Then a new String is created and != may be true, while the contents of two strings objects are the same. Using equals() method seems to be a preferred approach when comparing values of the string objects. You may find more info at https://stackoverflow.com/questions/513832/how-do-i-compare-strings-in-java